spmd/__init__.py

```diff
@@ -84,3 +84,14 @@ def replicate_module_params_buffers(m: nn.Module, mesh: DeviceMesh) -> None:
         )
 
     return module
+
+
+# All public APIs from spmd package
+__all__ = [
+    "DeviceMesh",
+    "Schema",
+    "SPMD",
+    "distribute_module",
+    "Shard",
+    "Replicate",
```

Comment:
What is the reason for having Shard and Replicate exposed both under spmd and spmd.dtensor?

Reply:
I think that's because user facing API needs to construct a `Schema` in SPMD, and user need to use `Shard` and `Replicate`? But it's a good point, maybe user can directly import from `spmd.tensor` for those placement types. Let me remove these placement APIs from spmd for now.